app/src/main/java/com/duckduckgo/app/di/AppComponent.kt

```diff
@@ -96,6 +99,12 @@ interface AppComponent : AndroidInjector<DuckDuckGoApplication> {
 
     fun inject(searchWidget: SearchWidget)
 
+    fun inject(searchAndFavsWidget: SearchAndFavoritesWidget)
+
+    fun inject(favoritesWidgetItemFactory: FavoritesWidgetService.FavoritesWidgetItemFactory)
+
+    fun inject(emptyFavoritesWidgetItemFactory: EmptyFavoritesWidgetService.EmptyFavoritesWidgetItemFactory)
+
```

Comment:
Why do we need these?  For instance for the `SearchAndFavoritesWidget` why can't we do the injection in `onReceive`?

BTW I think we also don't need the `fun inject(searchWidget: SearchWidget)`

Reply:
@aitorvs and I talked about this, and try directly using AndroidInjector, however it will not work without providing an `AndroidInjector.Factory` in the same way we do for each Activity. We decided to keep it as it is since we will remove dagger-android soon.